diff --git a/com/win32comext/shell/src/PyIAsyncOperation.h b/com/win32comext/shell/src/PyIAsyncOperation.h
index 1122609..a329738 100644
--- a/com/win32comext/shell/src/PyIAsyncOperation.h
+++ b/com/win32comext/shell/src/PyIAsyncOperation.h
@@ -4,6 +4,11 @@
 //
 // Interface Declaration
 
+#if _MSC_VER >= 1900
+#define IAsyncOperation IDataObjectAsyncCapability
+#define IID_IAsyncOperation IID_IDataObjectAsyncCapability
+#endif
+
 class PyIAsyncOperation : public PyIUnknown
 {
 public:
